### ISSUE TITLE
update request body or query in alias-level middleware not work

### ISSUE THREAD

**Issue by XHMM** [moleculerjs/moleculer-web#223]
When route-level middleware changed request body or query, action can get changed data, but if alias-level middleware changed it, it not work.

Here is a [reproduce demo](https://codesandbox.io/s/moleculer-api-routing-forked-1wi19): after run `npm run dev`, then run `curl -X POST localhost:8080/test`, you can find the (maybe) bug.

**Comment by icebob** [comment]
The reason is that the "query", "body" and "params" are processed after route middleware. So before action middlewares called, they are merged into `req.$params`, so you can modify through it, it will be the `ctx.params`.

**Comment by XHMM** [comment]
Thanks for reply, with the help of view source code, I understood the internal. 

I think the document description is not clear, which can make people mistakenly think that the use of the two middleware is the same, maybe should point out the difference?

**Comment by XHMM** [comment]
Thanks for reply, with the help of view source code, I understood the internal. 

I think the document description is not clear, which can make people mistakenly think that the use of the two middleware is the same, maybe should point out the difference?

**Comment by icebob** [comment]
Yeah, maybe the documentation needs some improvements to this topic.
